### PR TITLE
Convert MenuDroppoComponent component to an ES6 class

### PR DESCRIPTION
**Pull request by whymarrh**
Refs #7766

This PR converts the `MenuDroppoComponent` component to an ES6 class.

### CODE REVIEW

File: ui/app/components/app/menu-droppo.js [hunk 1]
```diff
@@ -1,33 +1,92 @@
...
+export default class MenuDroppoComponent extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool.isRequired,
+    innerStyle: PropTypes.object,
+    children: PropTypes.node.isRequired,
+    onClickOutside: PropTypes.func.isRequired,
+    containerClassName: PropTypes.string,
+    zIndex: PropTypes.number,
+    style: PropTypes.object.isRequired,
+    useCssTransition: PropTypes.bool.isRequired,
```
Comment (Gudahtt) [inline ui/app/components/app/menu-droppo.js]:
```suggestion
    useCssTransition: PropTypes.bool,
```

This should not be required - this prop is not always provided. It's passed through via the `dropdown` component, but that component is not passed this prop in at least one place (`ui/app/components/app/dropdowns/network-dropdown.js`)

File: ui/app/components/app/menu-droppo.js [hunk 2]
```diff
@@ -1,33 +1,92 @@
...
+
+    if (isOpen) {
+      this.outsideClickHandler = onClickOutside
+    } else if (isOpen) {
+      this.outsideClickHandler = null
+    }
+  }
+
+  componentDidMount = function () {
+    if (this && document.body) {
```
Comment (Gudahtt) [inline ui/app/components/app/menu-droppo.js]: Well, this is a weird thing to check for

**Review by Gudahtt — CHANGES_REQUESTED** [review]

File: ui/app/components/app/menu-droppo.js (hunk 2 above)
Comment (whymarrh) [reply ui/app/components/app/menu-droppo.js]: Yep

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**